src/fileparse.py

```diff
@@ -0,0 +1,66 @@
+import csv
+import logging
+from pathlib import Path
+from typing import Any, List, TextIO, Tuple, Union, Callable, Dict, cast
+
+
+def parse_csv(
+    file_path_or_buffer: Union[str, Path, TextIO],
+    select: List[str] = None,
+    types: List[Callable] = None,
+    has_headers: bool = True,
+    delimiter: str = ",",
+    silence_errors=False,
+) -> List[Any]:
+    """
+    Parse a csv file into a list.
+
+    :param file_path_or_buffer: Path to the csv file or file-like object
+    :param select: Columns to select
+    :param types: List of types to convert each column
+    :param has_headers: Indicator if file has headers.
+                        If False, creates a list of tuples otherwise a list of dictionaries
+    :param delimiter: Type of delimiter. Defaults to ','
+    :raises RuntimeError: Raises RuntimeError if both select and has_headers is False.
+    :return: List of items in the file
+    """
+    if not has_headers and select:
+        raise RuntimeError("select argument requires column headers")
+
```

Comment:
Maybe you can add 1 additional check: what happen if `types` and `select` are both provided bu they don't have the same length? I think it should raise an error

Linked to `types`, in case `has_headers` is set to `True` should we provide the `types` as a dict `column_name -> convert_fn` instead of a list?

Reply:
I agree with the `types` and `select` error. I can try add in `types` need to be a `dict` if there's headers.

Reply:
If `select` isn't given but there are headers then should `types` still work as a list?

Reply:
I think when you have `headers` in the CSV you should support both structures for `types`:
- list: it will follow the columns order
- dict: column_name -> convert_fn

Is it OK for you?

Reply:
That is what I ended up doing.